[CS2113-T15-3] Module Manager

RenzoTsai opened the pull request:
@bennychanya @chengTzeNing @DeetoMok

cheongisabella commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,123 @@
+user
+
+The following sequence diagram shows how the `Add to Semester` operation works:
+![Sequence Diagram of Add to Semester](https://raw.githubusercontent.com/RenzoTsai/tp/master/docs/UML%20img%20folder/Sequence%20Diagram%20of%20Add%20to%20Semester.png)
```
![Screenshot 2020-03-30 at 4 59 58 PM](https://user-images.githubusercontent.com/42826419/77895011-b54cc180-72a8-11ea-85cb-895c0e4db7c2.png)
Some remarks on your Add To Semester sequence diagram.

RenzoTsai replied:
Thanks for the feedbacks

cheongisabella commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,123 @@
+`(AddToSemCommand#excuted(SemesterList semesterList, AvailableModulesList availableModulesList))`
+
+##### Step 3:
+`AddToSemCommand#excuted()` then calls self method `AddToSemCommand#addModule()`.`AddToSemCommand#addModule()`
```
In the sequence diagram, the method name is "execute()" instead of "excuted()". Would it be better to use the same name?

RenzoTsai replied:
Sorry, I made a mistake here. Thanks!

cheongisabella commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,123 @@
+
+##### Step 3:
+`AddToSemCommand#excuted()` then calls self method `AddToSemCommand#addModule()`.`AddToSemCommand#addModule()`
+ then calls `AddToSemCommand#checkModuleExist(semesterList)` to check whether the selected 
```
The "checkModuleExitst()" method is not reflected in the sequence diagram. Would it be more comprehensible to add it in?

RenzoTsai replied:
Okay, thanks!

cheongisabella commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,123 @@
+2) Added using `add id/ID s/SEMESTER mc/MODULE_CREDIT` command
+
+Step 2:
+User executes `CAP` command to view his own CAP. The `CAP` commands is parsed through `Parser`, which would then return 
```
![Screenshot 2020-03-30 at 5 16 42 PM](https://user-images.githubusercontent.com/42826419/77896285-9f400080-72aa-11ea-8261-1469bd2cde65.png)
Some remarks to the Calculate CAP operation sequence diagram.

cheongisabella commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,123 @@
+module is already in the selected module list (which is`semesterList`). If the module is not in the list, 
+`AddToSemCommand#addModule()` will check whether there is a semester list whose name is the module's semester name. If 
+the semester list exist, the module will be added to the list. If not, `AddToSemCommand#addModule()` will create a new 
+semester list and then add this module to the new list.
```
This part about the list seems a bit confusing. Would it be good to explain the different lists and identify the different lists in this section more explicitly?

RenzoTsai replied:
OKay!

cheongisabella reviewed (commented):
Here are some comments and suggestions on the developer guide. Good job on the overall format of the developer guide, which seems good and neat. The explanations are also concise.

g-lilian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,172 @@
+
+##### Step 1:
+The user launches the application for the first time. The `SemesterList` will be initialized with the none 
+`SemModulesList` in it.
```
Do you mean to say "empty SemModulesList"?

g-lilian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,172 @@
+(i.e:`semesterModulesList`, which is a `ArrayList<SelectedModule>`) whose name is the module's semester name. 
+If the semester list exist, the module will be added to the list. 
+If not, `AddToSemCommand#addModule()` will create a new semester list and then add this module to the new list. and the
+the new semester list will be added to `semesterList` as well.
```
I found this section quite confusing to read as there are a few variables and classes with similar names (semesterList, SemModulesList, semesterModulesList, SemesterList), and the multiple 'if' condition checks. Perhaps it would help to briefly explain the connection between semesterList and semesterModulesList, at the very beginning, before delving into the logic? I.e. "semesterList represents XXX and stores instances of SemModuleList. semesterModulesList represents XXX and stores XXX"

g-lilian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,172 @@
+user
+
+The following sequence diagram shows how the `Add to Semester` operation works:
+![Sequence Diagram of Add to Semester](https://github.com/RenzoTsai/tp/blob/Update_DG/docs/UML%20img%20folder/Sequence%20Diagram%20of%20Add%20to%20Semester.png)
```
Some comments on the sequence diagram
![Screenshot_4](https://user-images.githubusercontent.com/50647023/77994380-809a4200-735c-11ea-8b27-0a33cd5998c0.jpg)

g-lilian replied:
On a side note, the diagram doesn't show up properly when I load the website, and stays like this when I refresh. I had to right click the icon and open in a new tab to view it. Could there be a problem with the file linking?
![Screenshot_1](https://user-images.githubusercontent.com/50647023/77994637-f0a8c800-735c-11ea-88c8-8856f7df5bfa.jpg)

g-lilian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,172 @@
+with a pattern of `#.00`. `Ui.showcap(cap)` is called to display the user's cap using the formatted `String`.
+
+The following diagram shows how the Calculate CAP operation works:
+![Calculate CAP feature](https://github.com/bennychanya/tp/blob/master/CalculateCap.png?raw=true)
```
I agree with what Isabella said about the construction of CalculateCapCommand. Otherwise it looks good to me. Perhaps you may want to specify the return value (grade) of the getGrade() method to be consistent, as you did so for totalCap?

alaukiknpant commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,123 @@
+user
+
+The following sequence diagram shows how the `Add to Semester` operation works:
+![Sequence Diagram of Add to Semester](https://raw.githubusercontent.com/RenzoTsai/tp/master/docs/UML%20img%20folder/Sequence%20Diagram%20of%20Add%20to%20Semester.png)
```
Can it be the case that steps also refer tot the Sequence Diagram while explaining the `Add To Semester` feature?

g-lilian reviewed (commented):
Overall, the DG looks well-organised and informative. Some areas can be clearer and the sequence diagrams may need a bit of tweaking.

alaukiknpant commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,123 @@
+
+Step 2:
+User executes `CAP` command to view his own CAP. The `CAP` commands is parsed through `Parser`, which would then return 
+`CalculateCapCommand()`. `CalculateCapCommand.execute()` is then called.
```
![Screen Shot 2020-03-31 at 14 50 32](https://user-images.githubusercontent.com/40881096/77995866-22bb2980-735f-11ea-84df-cc59df1be880.png)

I am wondering if it is fully clear weather object representing CalculateCapCommand class or a Person Object Calculates the total cap. More specifically, the setTotalCap() method and getTotalCap() methods are pointed in the same direction, so I wasn't a 100% sure from the diagram where the CAP calculation was happening.

alaukiknpant commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,123 @@
+The `Model` component
+* Stores a `Person` object that represents the user's details. (e.g. `totalCap`, `totalModuleCreditCompleted`)
+* Stores the `ModuleList` and `SemesterList` Data
+* Does not depend on any of the other three components
```
I was wondering if Model component is not related to other three components, then how would a Person object, which is stored in Model Component, have their CAP calculated? Maybe "does not depend" is not the correct phrase here?

alaukiknpant commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,123 @@
+
+### Implementation
+
+#### `Add to Semester` feature 
```
The diagram looks very neat and helps the developer understand the basic logic of add to semester future. So perhaps, is the explanation repetitive here?

RenzoTsai committed:
Fix a bug in CAP (will get a wrong CAP if involves S/U)

DeetoMok committed:
Merge pull request #114 from RenzoTsai/Fix_Bugs

Fix some bugs

Commit:
fixed bugs like creating exception for unacceptable values for modular credits and semester values

Commit:
added ClearCommand and fixed code to make module codes and grades uppercase

Commit:
fixed for checkstyle

Commit:
fixed bug for when view commands are executed even when the wrong formats are used

RenzoTsai committed:
Update code to support showing preReqs which are not in available modules list

RenzoTsai committed:
Fix some code style bugs

Commit:
added Ui message when module plan is empty

Commit:
remove checkstyle error

RenzoTsai committed:
Fix a bug when add the same modules to available modules list

Commit:
Updated Developer Guide

Commit:
Updated Developer Guide

Commit:
Updated Developer Guide

DeetoMok committed:
Merge pull request #117 from DeetoMok/master

Updated Developer Guide

DeetoMok committed:
Merge pull request #116 from RenzoTsai/Fix_Bugs

Allow to show prerequisites which are not in the available modules list and fix a bug when add same modules to AML

Commit:
Updated Developer Guide and Architecture.png

DeetoMok committed:
Merge pull request #118 from DeetoMok/master

Updated Developer Guide and Architecture.png

Commit:
Added object diagram for logic component and Object_Diagram_of_Logic_Component.png

Commit:
Updated object diagram for logic component and Object_Diagram_of_Logic_Component.png

bennychanya committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

Commit:
Uploaded Class_Diagram_of_Model_Component.png

DeetoMok committed:
Merge pull request #119 from DeetoMok/master

updated DG and added Class Diagram of Model Component

Commit:
Updated Developer Guide

DeetoMok committed:
Merge pull request #120 from DeetoMok/master

Updated Developer Guide

Commit:
Updated Developer Guide and renamed AddToDataCommand.java to AddToAvailableCommand.java

Commit:
Fixed a delete bug (could not delete module when uppercase letter and lowercase letter different)

Commit:
Fixed a add module bug (module added did not convert lowercase module code to uppercase module code)

Commit:
fixed bug (could not delete module in available module list if only difference was upper/lowercase related)

Commit:
updated DG

DeetoMok committed:
Merge pull request #115 from DeetoMok/FixBug

Fix bug, added ClearCommand and fixed code to make module codes and grades uppercase

Commit:
updated UI HELP String

Commit:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

# Conflicts:
#	data/availableModulesList.csv
#	data/semesterList.csv
#	src/main/java/seedu/duke/ui/Ui.java
#	text-ui-test/EXPECTED.TXT

Commit:
reduce checkstyle error

RenzoTsai committed:
Show "Please input your command:"

RenzoTsai committed:
Update text-ui-test

RenzoTsai committed:
Merge pull request #121 from DeetoMok/master

Fixed bugs, updated user guide and developer guide

RenzoTsai committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp into Fix_Bugs

RenzoTsai committed:
Update UG

RenzoTsai committed:
Merge pull request #122 from RenzoTsai/Fix_Bugs

Show "Please input your command:" and update UG

RenzoTsai committed:
Fix a output error.

RenzoTsai committed:
Fix a output error

RenzoTsai committed:
Merge pull request #123 from RenzoTsai/master

Fix an output error

tzzzening committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

tzzzening committed:
Add Sequence Diagrams for DG

tzzzening committed:
Update Sequence Diagrams in DG

tzzzening committed:
Edit sequence diagram for DeleteFromSem

tzzzening committed:
Add description for DeleteFromAvailableCommand

tzzzening committed:
Edit minor text

tzzzening committed:
Make minor changes to AvailableModulesList

RenzoTsai committed:
Update Runze's PPP

RenzoTsai committed:
Merge pull request #125 from RenzoTsai/Runze-PPP

Update Runze's PPP

tzzzening committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

tzzzening committed:
Add chengtzening_PPP

Commit:
added JUnitTest and some Javadoc

Commit:
reduced checkstyle error

DeetoMok committed:
Merge pull request #127 from DeetoMok/JUnitTest

Added J unit test and some JavaDoc

RenzoTsai committed:
Merge pull request #126 from chengTzeNing/edit-tzening-PPP

Add chengtzening PPP

RenzoTsai committed:
Merge pull request #124 from chengTzeNing/edit-DG

Edit DG

RenzoTsai committed:
Fix bugs in DG's graph showing

RenzoTsai committed:
Merge pull request #128 from RenzoTsai/master

Fix bugs in DG's graph showing

RenzoTsai committed:
Update DG's graph

RenzoTsai committed:
Merge pull request #129 from RenzoTsai/master

Update DG's graph

Commit:
update

RenzoTsai committed:
Fix some bugs

RenzoTsai committed:
Update storage in DG

RenzoTsai committed:
Update DeveloperGuide.md

RenzoTsai committed:
Update renzotsai.md

RenzoTsai committed:
Update renzotsai.md

RenzoTsai committed:
Update renzotsai.md

RenzoTsai committed:
Update renzotsai.md

RenzoTsai committed:
Merge pull request #130 from RenzoTsai/UpdateDG_UG

Update DG&UG

RenzoTsai committed:
Update UserGuide.md

RenzoTsai committed:
Update PPP

RenzoTsai committed:
Merge pull request #131 from RenzoTsai/UpdateDG_UG

Update PPP

RenzoTsai committed:
Update PPP

RenzoTsai committed:
Merge pull request #132 from RenzoTsai/UpdateDG_UG

Update RUNZE's PPP

Commit:
update darylmok-PPP

Commit:
updated DeveloperGuide.md Manual Testing section

Commit:
added table of contents in DeveloperGuide.md

Commit:
updated UserGuide.md

DeetoMok committed:
update DeveloperGuide.md Table of Contents

DeetoMok committed:
Added table of contents in User Guide

DeetoMok committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

 Conflicts:
	docs/UserGuide.md

DeetoMok committed:
Merge pull request #133 from DeetoMok/master

Created Table of Contents for Developer Guide

DeetoMok committed:
updated PPP

tzzzening committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

RenzoTsai committed:
Update JavaDoc

RenzoTsai committed:
Merge pull request #134 from RenzoTsai/AddJavaDoc

Update JavaDoc

RenzoTsai committed:
Update UserGuide.md

RenzoTsai committed:
Update UserGuide.md

tzzzening committed:
Add Sequence Diagram for FindCommand

tzzzening committed:
Add FindCommand to DG

tzzzening committed:
Delete Sequence Diagram for DeleteFromSemCommand

tzzzening committed:
Delete this Sequence Diagram

tzzzening committed:
Add updated Sequence Diagram for DeleteFromSemCommand

tzzzening committed:
Add DeleteFromSemCommand Sequence Diagram to DG

DeetoMok committed:
added MarkAsDone Sequence Diagram

DeetoMok committed:
Merge pull request #135 from DeetoMok/master

added MarkAsDone Sequence Diagram

DeetoMok committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp into darylmok-PPP

 Conflicts:
	docs/DeveloperGuide.md

tzzzening committed:
Add FindCommand to content page for UG and DG

tzzzening committed:
Merge branch 'master' into DG-and-UG

tzzzening committed:
Merge pull request #136 from chengTzeNing/DG-and-UG

Edit DG and UG

tzzzening committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

RenzoTsai committed:
Update DeveloperGuide.md

DeetoMok committed:
updated TOC format

tzzzening committed:
Shorten PPP

tzzzening committed:
Merge pull request #4 from DeetoMok/master

Format for making new line in Markdown

tzzzening committed:
Merge branch 'master' of https://github.com/chengTzeNing/tp

tzzzening committed:
Edit headings in PPP

tzzzening committed:
Edit to make it visible in .md file

tzzzening committed:
Merge pull request #137 from chengTzeNing/tzening-PPP

Tzening ppp

tzzzening committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

DeetoMok committed:
updated deetomok.md

DeetoMok committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

DeetoMok committed:
corrected format of TOC for UserGuide.md

bennychanya committed:
Refactor out boolean condition

bennychanya committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

bennychanya committed:
Merge branch 'master' of https://github.com/bennychanya/tp into RefactorCodeV2

tzzzening committed:
Update AboutUS.md

tzzzening committed:
Merge pull request #138 from chengTzeNing/add-picture

Add picture

tzzzening committed:
Merge pull request #5 from DeetoMok/master

Update the Format for Userguide

DeetoMok committed:
updated deetomok.md

DeetoMok committed:
Merge branch 'darylmok-PPP'

bennychanya committed:
Changed CalculateCap_UML_diagram

bennychanya committed:
Refactor if-statement condition to be clearer

bennychanya committed:
Edited Check Method to allow Module to be added if user failed the selected module
Changed condition for thrown exception to allow 0 MC module
Changed condition for thrown exception to allow modules up to 20MC

DeetoMok committed:
Added find in command summary of user guide

bennychanya committed:
Edited Javadoc for markAsDoneCommand

DeetoMok committed:
Merge pull request #140 from DeetoMok/master

Added DeetoMok-PPP

bennychanya committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

bennychanya committed:
Merge branch 'master' of https://github.com/bennychanya/tp into RefactorCodeV2

tzzzening committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

tzzzening committed:
Merge branch 'master' of https://github.com/chengTzeNing/tp

bennychanya committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

tzzzening committed:
Edit FindCommand in Ui

RenzoTsai committed:
Update EXPECTED.txt

tzzzening committed:
Merge pull request #141 from chengTzeNing/random-edits-1

Random edits 1

tzzzening committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

bennychanya committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

RenzoTsai committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

* 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp:
  Edit FindCommand in Ui

bennychanya committed:
Merge branch 'master' of https://github.com/bennychanya/tp into RefactorCodeV2

RenzoTsai committed:
Fix a bug in find

RenzoTsai committed:
Fix a code style error

RenzoTsai committed:
Merge pull request #143 from RenzoTsai/FixBugOfFind

Fix bugs of find

RenzoTsai committed:
Merge branch 'RefactorCodeV2' of https://github.com/bennychanya/tp into Revise-Refactor-code-v2

* 'RefactorCodeV2' of https://github.com/bennychanya/tp:
  Edited Javadoc for markAsDoneCommand
  Edited Check Method to allow Module to be added if user failed the selected module Changed condition for thrown exception to allow 0 MC module Changed condition for thrown exception to allow modules up to 20MC
  Refactor if-statement condition to be clearer
  Changed CalculateCap_UML_diagram
  Refactor out boolean condition
  Edited UML diagram
  Changed UML diagram for DG
  Added UML Diagram for Calculate CAP
  Added UML diagram for CalculateCap
  Added Calculate Cap Implementation into DeveloperGuide
  Update src/main/java/seedu/duke/command/CalculateCapCommand.java

RenzoTsai committed:
Fix a bug

bennychanya committed:
Fixed Check method

bennychanya committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

bennychanya committed:
Merge branch 'master' of https://github.com/bennychanya/tp into RefactorCodeV2

RenzoTsai committed:
Merge branch 'RefactorCodeV2' of https://github.com/bennychanya/tp into Revise-Refactor-code-v2

* 'RefactorCodeV2' of https://github.com/bennychanya/tp:
  Fixed Check method

RenzoTsai committed:
Fix a bug

RenzoTsai committed:
Update UG

RenzoTsai committed:
Fix a bug involves in view /cc  delete and markAsDone

RenzoTsai committed:
Update data's file

RenzoTsai committed:
Update test

RenzoTsai committed:
Update UG

RenzoTsai committed:
Update delete from semester list

bennychanya committed:
Edited Javadoc for CalculateCapCommand, DeleteFromSemCommand, FindCommand, ViewCommand

bennychanya committed:
Removed type from DeleteFromSem as it has no uses

bennychanya committed:
Edited Javadoc to pass CLI

RenzoTsai committed:
Update UG

bennychanya committed:
Merge pull request #3 from RenzoTsai/Revise-Refactor-code-v2

Revise refactor code v2

bennychanya committed:
Reserve changed to allow duplicate module if module has failed

bennychanya committed:
Merge branch 'RefactorCodeV2' of https://github.com/bennychanya/tp into RefactorCodeV2

# Conflicts:
#	src/main/java/seedu/duke/command/AddToSemCommand.java

bennychanya committed:
Edited code indentation to pass CLI test

tzzzening committed:
Merge pull request #144 from RenzoTsai/master

Fix bugs about delete, markAsDone and view cc

bennychanya committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

bennychanya committed:
Merge Conflict

bennychanya committed:
Edit Expected.txt to passed CLI

bennychanya committed:
Edited AboutUs

bennychanya committed:
Changed comparator "equals" to "equalsIgnorecase" within AddToSemCommand and MarkAsDoneCommand

RenzoTsai committed:
Update renzotsai.md

RenzoTsai committed:
Update renzotsai.md

RenzoTsai committed:
Merge pull request #139 from bennychanya/RefactorCodeV2

Refactor code v2

bennychanya committed:
Merge branch 'master' of https://github.com/bennychanya/tp into Duplicate_Failed_Module

RenzoTsai committed:
Fix a bug in text ui test

RenzoTsai committed:
Merge pull request #145 from RenzoTsai/master

Fix a little error in text ui test

DeetoMok committed:
Updated deetomok.md

DeetoMok committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

 Conflicts:
	data/semesterList.csv
	text-ui-test/EXPECTED.TXT
	text-ui-test/data/semesterList.csv
	text-ui-test/input.txt

bennychanya committed:
Edit to decimalformat to display "0.00" if CAP = 0

bennychanya committed:
Added function to input placeholder module for failed module

RenzoTsai committed:
Merge branch 'Duplicate_Failed_Module' of https://github.com/bennychanya/tp into Revise-Duplicate-failed-module

* 'Duplicate_Failed_Module' of https://github.com/bennychanya/tp:
  Added function to input placeholder module for failed module
  Edit to decimalformat to display "0.00" if CAP = 0
  Edited AboutUs

RenzoTsai committed:
Fix a bug when remark a module from A to F/SU

RenzoTsai committed:
Update UG

bennychanya committed:
Edited UserGuide to include F or CU grade scenario

bennychanya committed:
Merge pull request #4 from RenzoTsai/Revise-Duplicate-failed-module

Revise duplicate failed module

RenzoTsai committed:
Update UG

RenzoTsai committed:
Merge pull request #147 from RenzoTsai/Revise-UG

Update UG

bennychanya committed:
Merge branch 'master' of https://github.com/bennychanya/tp into Duplicate_Failed_Module

DeetoMok committed:
Merge pull request #146 from bennychanya/Duplicate_Failed_Module

Duplicate failed module

DeetoMok committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

 Conflicts:
	text-ui-test/EXPECTED.TXT

bennychanya committed:
Updated UG for MarkAsDone

RenzoTsai committed:
Fix a bug in clear

bennychanya committed:
Merge pull request #148 from RenzoTsai/master

Fix a bug in clear

bennychanya committed:
Edited my own PPP

bennychanya committed:
Merge pull request #149 from bennychanya/Duplicate_Failed_Module

Edit to fix UG table content

RenzoTsai committed:
Update comments in markAsDone

bennychanya committed:
Minor edit to Benny's PPP

DeetoMok committed:
created appendFailString method in MarkAsDoneCommand

DeetoMok committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

 Conflicts:
	docs/UserGuide.md

bennychanya committed:
Merge pull request #151 from bennychanya/Duplicate_Failed_Module

Minor edit to Benny's PPP

bennychanya committed:
Merge pull request #150 from RenzoTsai/master

Update comments in markAsDone

tzzzening committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

DeetoMok committed:
updated MarkAsDone Sequence diagram, and added appendFailString method

DeetoMok committed:
Updated deetomok.md

DeetoMok committed:
Merge branch 'master' into master

DeetoMok committed:
Added Javadoc

DeetoMok committed:
updated clearer Mark_As_Done_Sequence_Diagram.png

DeetoMok committed:
Merge pull request #152 from DeetoMok/master

Updated MarkAsDone UML and created appendFailString method in MarkAsDone

DeetoMok committed:
removed unclear boolean variables

DeetoMok committed:
updated DG

DeetoMok committed:
update DG

DeetoMok committed:
Merge pull request #153 from DeetoMok/master

Updated DG Model Component Class Diagram

bennychanya committed:
Updated DG for Mark As Done Command

DeetoMok committed:
updated readme and DG

DeetoMok committed:
Merge pull request #155 from DeetoMok/master

updated readme and DG

bennychanya committed:
Merge pull request #154 from bennychanya/master

Updated DG for Mark As Done Command

DeetoMok committed:
added viewing modules in DeveloperGuide.md

RenzoTsai committed:
Update renzotsai.md

DeetoMok committed:
updated viewing modules in DeveloperGuide.md

DeetoMok committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T15-3/tp

bennychanya committed:
Updated DG with design consideration for Mark As Done

DeetoMok committed:
updated deetomok.md

DeetoMok committed:
updated deetomok.md

bennychanya committed:
Merge pull request #157 from bennychanya/master

Updated DG with design consideration for Mark As Done

DeetoMok committed:
updated DG

DeetoMok committed:
Merge pull request #156 from DeetoMok/master

Added Viewing in DG and design considerations

RenzoTsai committed:
Update add to sem design consideration in DG

bennychanya committed:
Update Benny's PPP

RenzoTsai committed:
Update DG &PPP

tzzzening committed:
Add design considerations for AvailableModulesList

bennychanya committed:
Merge pull request #159 from bennychanya/master

Update Benny's PPP

tzzzening committed:
Merge branch 'master' into random-edits-2

RenzoTsai committed:
Merge pull request #160 from chengTzeNing/random-edits-2

Add design considerations

tzzzening committed:
Merge pull request #158 from RenzoTsai/master

Update add‘s design consideration in DG

bennychanya committed:
Added table for Design consideration

DeetoMok committed:
updated deetomok.md

DeetoMok committed:
Merge pull request #162 from DeetoMok/master

updated deetomok.md

DeetoMok committed:
updated deetomok.md

bennychanya committed:
Merge pull request #161 from bennychanya/master

Added table for Design consideration

DeetoMok committed:
Merge pull request #163 from DeetoMok/master

updated deetomok.md

DeetoMok committed:
update AboutUs.md

DeetoMok committed:
Merge pull request #164 from DeetoMok/master

update AboutUs.md

DeetoMok committed:
update deetomok.md

DeetoMok committed:
Merge pull request #165 from DeetoMok/master

update deetomok.md

DeetoMok committed:
update deetomok.md

DeetoMok committed:
Merge pull request #166 from DeetoMok/master

update deetomok.md

DeetoMok committed:
update UG

DeetoMok committed:
Merge pull request #167 from DeetoMok/master

update UG
